### ISSUE TITLE
An error in handling requests datatables

### ISSUE THREAD

**Issue by dead23angel** [aacerox/node-rest-client#131]
I using node-rest-client in express.

route

``` js
app.get('/groups', middleware.isAjax, function(req, res) {
    client.methods.Groups(client.setData(userReq), function(data, response) {
            if(! ClientFactory.api.checkResponse(userReq, response)) {
                res.redirect('/logout');
                return;
            }

            res.send(data);
        });
});
```

initialRestClient.js

``` js
var Client = require('node-rest-client').Client,
    client = new Client();

var appMode = process.env.NODE_ENV || 'local';
var config = require('./config/'+ appMode +'.js');

var methodsInit = function() {
    client.registerMethod("Auth", config.api_url + "login", "POST");
    client.registerMethod('ForceRefreshToken', config.api_url + "refresh", 'GET');
    client.registerMethod("Groups", config.api_url + "groups", 'GET');
};

var isEmptyObject = function (obj) {
    return Object.keys(obj).length === 0;
};

var setData = function(req, path) {
    var dataClient = {
        headers: {
            "Server-token": config.secret_client_key
        },
        requestConfig: {
            timeout: 100,
            noDelay: true,
            keepAlive: true,
            keepAliveDelay: 60
        },
        responseConfig: {
            timeout: 100
        },
        parameters: {

        }
    };

    if(typeof path)
        dataClient.path = path;

    if(! isEmptyObject(req.query)) {
        dataClient.parameters = req.query;
    } else if(! isEmptyObject(req.body)) {
        dataClient.parameters = req.body;
    }

    return dataClient;
};
```

datatables sent request to route
request

```
/groups?draw=1&columns%5B0%5D%5Bdata%5D=0&columns%5B0%5D%5Bname%5D=&columns%5B0%5D%5Bsearchable%5D=true&columns%5B0%5D%5Borderable%5D=true&columns%5B0%5D%5Bsearch%5D%5Bvalue%5D=&columns%5B0%5D%5Bsearch%5D%5Bregex%5D=false&columns%5B1%5D%5Bdata%5D=1&columns%5B1%5D%5Bname%5D=&columns%5B1%5D%5Bsearchable%5D=true&columns%5B1%5D%5Borderable%5D=true&columns%5B1%5D%5Bsearch%5D%5Bvalue%5D=&columns%5B1%5D%5Bsearch%5D%5Bregex%5D=false&columns%5B2%5D%5Bdata%5D=2&columns%5B2%5D%5Bname%5D=&columns%5B2%5D%5Bsearchable%5D=true&columns%5B2%5D%5Borderable%5D=true&columns%5B2%5D%5Bsearch%5D%5Bvalue%5D=&columns%5B2%5D%5Bsearch%5D%5Bregex%5D=false&columns%5B3%5D%5Bdata%5D=3&columns%5B3%5D%5Bname%5D=&columns%5B3%5D%5Bsearchable%5D=true&columns%5B3%5D%5Borderable%5D=true&columns%5B3%5D%5Bsearch%5D%5Bvalue%5D=&columns%5B3%5D%5Bsearch%5D%5Bregex%5D=false&order%5B0%5D%5Bcolumn%5D=0&order%5B0%5D%5Bdir%5D=asc&start=0&length=10&search%5Bvalue%5D=&search%5Bregex%5D=false&_=1475380958097
```

node-rest-client sent request to api
request

```
[2] [dev:server] IncomingMessage {
[2]   _readableState: 
[2]    ReadableState {
[2]      objectMode: false,
[2]      highWaterMark: 16384,
[2]      buffer: BufferList { head: null, tail: null, length: 0 },
[2]      length: 0,
[2]      pipes: null,
[2]      pipesCount: 0,
[2]      flowing: true,
[2]      ended: true,
[2]      endEmitted: true,
[2]      reading: false,
[2]      sync: false,
[2]      needReadable: false,
[2]      emittedReadable: false,
[2]      readableListening: false,
[2]      resumeScheduled: false,
[2]      defaultEncoding: 'utf8',
[2]      ranOut: false,
[2]      awaitDrain: 0,
[2]      readingMore: false,
[2]      decoder: null,
[2]      encoding: null },
[2]   readable: false,
[2]   domain: null,
[2]   _events: 
[2]    { end: [ [Function: responseOnEnd], [Function] ],
[2]      timeout: [Function],
[2]      data: [Function],
[2]      error: [Function] },
[2]   _eventsCount: 4,
[2]   _maxListeners: undefined,
[2]   socket: 
[2]    Socket {
[2]      connecting: false,
[2]      _hadError: false,
[2]      _handle: null,
[2]      _parent: null,
[2]      _host: 'localhost',
[2]      _readableState: 
[2]       ReadableState {
[2]         objectMode: false,
[2]         highWaterMark: 16384,
[2]         buffer: [Object],
[2]         length: 0,
[2]         pipes: null,
[2]         pipesCount: 0,
[2]         flowing: true,
[2]         ended: true,
[2]         endEmitted: true,
[2]         reading: false,
[2]         sync: false,
[2]         needReadable: false,
[2]         emittedReadable: false,
[2]         readableListening: false,
[2]         resumeScheduled: false,
[2]         defaultEncoding: 'utf8',
[2]         ranOut: false,
[2]         awaitDrain: 0,
[2]         readingMore: false,
[2]         decoder: null,
[2]         encoding: null },
[2]      readable: false,
[2]      domain: null,
[2]      _events: 
[2]       { end: [Object],
[2]         finish: [Function: onSocketFinish],
[2]         _socketEnd: [Function: onSocketEnd],
[2]         free: [Function: onFree],
[2]         close: [Object],
[2]         agentRemove: [Function: onRemove],
[2]         drain: [Function: ondrain],
[2]         error: [Function: socketErrorListener],
[2]         data: [Function: socketOnData],
[2]         timeout: [Object] },
[2]      _eventsCount: 10,
[2]      _maxListeners: undefined,
[2]      _writableState: 
[2]       WritableState {
[2]         objectMode: false,
[2]         highWaterMark: 16384,
[2]         needDrain: false,
[2]         ending: true,
[2]         ended: true,
[2]         finished: true,
[2]         decodeStrings: false,
[2]         defaultEncoding: 'utf8',
[2]         length: 0,
[2]         writing: false,
[2]         corked: 0,
[2]         sync: false,
[2]         bufferProcessing: false,
[2]         onwrite: [Function],
[2]         writecb: null,
[2]         writelen: 0,
[2]         bufferedRequest: null,
[2]         lastBufferedRequest: null,
[2]         pendingcb: 0,
[2]         prefinished: true,
[2]         errorEmitted: false,
[2]         bufferedRequestCount: 0,
[2]         corkedRequestsFree: [Object] },
[2]      writable: false,
[2]      allowHalfOpen: false,
[2]      destroyed: true,
[2]      _bytesDispatched: 657,
[2]      _sockname: null,
[2]      _pendingData: null,
[2]      _pendingEncoding: '',
[2]      server: null,
[2]      _server: null,
[2]      parser: null,
[2]      _httpMessage: 
[2]       ClientRequest {
[2]         domain: null,
[2]         _events: [Object],
[2]         _eventsCount: 3,
[2]         _maxListeners: undefined,
[2]         output: [],
[2]         outputEncodings: [],
[2]         outputCallbacks: [],
[2]         outputSize: 0,
[2]         writable: true,
[2]         _last: true,
[2]         upgrading: false,
[2]         chunkedEncoding: false,
[2]         shouldKeepAlive: false,
[2]         useChunkedEncodingByDefault: false,
[2]         sendDate: false,
[2]         _removedHeader: [Object],
[2]         _contentLength: 0,
[2]         _hasBody: true,
[2]         _trailer: '',
[2]         finished: true,
[2]         _headerSent: true,
[2]         socket: [Circular],
[2]         connection: [Circular],
[2]         _header: 'GET /api/v1/groups?draw=1&columns=%5Bobject%20Object%5D&columns=%5Bobject%20Object%5D&columns=%5Bobject%20Object%5D&columns=%5Bobject%20Object%5D&order=%5Bobject%20Object%5D&start=0&length=10&search=%5Bobject%20Object%5D&_=1475381757531 HTTP/1.1\r\nServer-token: B@s~DSSwewaC63n0#xIf~9ROAoU$d$Z1WM|\r\nAuthorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwOi8vbG9jYWxob3N0OjgwMDAvYXBpL3YxL2xvZ2luIiwiaWF0IjoxNDc1MzgxNzU1LCJleHAiOjE0NzUzODUzNTUsIm5iZiI6MTQ3NTM4MTc1NSwianRpIjoiYTE2NjY3ODA4NzA0ZTljM2NmNmZiZjMyMDM4NGIxY2YiLCJzdWIiOjF9.tz_DUHUaaD77zMkAk6cUd7-hUQYoQHlr3WGgDQZAupA\r\nContent-Length: 0\r\nHost: localhost:8000\r\nConnection: close\r\n\r\n',
[2]         _headers: [Object],
[2]         _headerNames: [Object],
[2]         _onPendingData: null,
[2]         agent: [Object],
[2]         socketPath: undefined,
[2]         method: 'GET',
[2]         path: '/api/v1/groups?draw=1&columns=%5Bobject%20Object%5D&columns=%5Bobject%20Object%5D&columns=%5Bobject%20Object%5D&columns=%5Bobject%20Object%5D&order=%5Bobject%20Object%5D&start=0&length=10&search=%5Bobject%20Object%5D&_=1475381757531',
[2]         _ended: true,
[2]         timeoutCb: [Function: emitTimeout],
[2]         parser: null,
[2]         res: [Circular] },
[2]      _idleTimeout: -1,
[2]      _idleNext: null,
[2]      _idlePrev: null,
[2]      _idleStart: 55893,
[2]      read: [Function],
[2]      _consuming: true,
[2]      write: [Function: writeAfterFIN] },
[2]   connection: 
[2]    Socket {
[2]      connecting: false,
[2]      _hadError: false,
[2]      _handle: null,
[2]      _parent: null,
[2]      _host: 'localhost',
[2]      _readableState: 
[2]       ReadableState {
[2]         objectMode: false,
[2]         highWaterMark: 16384,
[2]         buffer: [Object],
[2]         length: 0,
[2]         pipes: null,
[2]         pipesCount: 0,
[2]         flowing: true,
[2]         ended: true,
[2]         endEmitted: true,
[2]         reading: false,
[2]         sync: false,
[2]         needReadable: false,
[2]         emittedReadable: false,
[2]         readableListening: false,
[2]         resumeScheduled: false,
[2]         defaultEncoding: 'utf8',
[2]         ranOut: false,
[2]         awaitDrain: 0,
[2]         readingMore: false,
[2]         decoder: null,
[2]         encoding: null },
[2]      readable: false,
[2]      domain: null,
[2]      _events: 
[2]       { end: [Object],
[2]         finish: [Function: onSocketFinish],
[2]         _socketEnd: [Function: onSocketEnd],
[2]         free: [Function: onFree],
[2]         close: [Object],
[2]         agentRemove: [Function: onRemove],
[2]         drain: [Function: ondrain],
[2]         error: [Function: socketErrorListener],
[2]         data: [Function: socketOnData],
[2]         timeout: [Object] },
[2]      _eventsCount: 10,
[2]      _maxListeners: undefined,
[2]      _writableState: 
[2]       WritableState {
[2]         objectMode: false,
[2]         highWaterMark: 16384,
[2]         needDrain: false,
[2]         ending: true,
[2]         ended: true,
[2]         finished: true,
[2]         decodeStrings: false,
[2]         defaultEncoding: 'utf8',
[2]         length: 0,
[2]         writing: false,
[2]         corked: 0,
[2]         sync: false,
[2]         bufferProcessing: false,
[2]         onwrite: [Function],
[2]         writecb: null,
[2]         writelen: 0,
[2]         bufferedRequest: null,
[2]         lastBufferedRequest: null,
[2]         pendingcb: 0,
[2]         prefinished: true,
[2]         errorEmitted: false,
[2]         bufferedRequestCount: 0,
[2]         corkedRequestsFree: [Object] },
[2]      writable: false,
[2]      allowHalfOpen: false,
[2]      destroyed: true,
[2]      _bytesDispatched: 657,
[2]      _sockname: null,
[2]      _pendingData: null,
[2]      _pendingEncoding: '',
[2]      server: null,
[2]      _server: null,
[2]      parser: null,
[2]      _httpMessage: 
[2]       ClientRequest {
[2]         domain: null,
[2]         _events: [Object],
[2]         _eventsCount: 3,
[2]         _maxListeners: undefined,
[2]         output: [],
[2]         outputEncodings: [],
[2]         outputCallbacks: [],
[2]         outputSize: 0,
[2]         writable: true,
[2]         _last: true,
[2]         upgrading: false,
[2]         chunkedEncoding: false,
[2]         shouldKeepAlive: false,
[2]         useChunkedEncodingByDefault: false,
[2]         sendDate: false,
[2]         _removedHeader: [Object],
[2]         _contentLength: 0,
[2]         _hasBody: true,
[2]         _trailer: '',
[2]         finished: true,
[2]         _headerSent: true,
[2]         socket: [Circular],
[2]         connection: [Circular],
[2]         _header: 'GET /api/v1/groups?draw=1&columns=%5Bobject%20Object%5D&columns=%5Bobject%20Object%5D&columns=%5Bobject%20Object%5D&columns=%5Bobject%20Object%5D&order=%5Bobject%20Object%5D&start=0&length=10&search=%5Bobject%20Object%5D&_=1475381757531 HTTP/1.1\r\nServer-token: B@s~DSSwewaC63n0#xIf~9ROAoU$d$Z1WM|\r\nAuthorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwOi8vbG9jYWxob3N0OjgwMDAvYXBpL3YxL2xvZ2luIiwiaWF0IjoxNDc1MzgxNzU1LCJleHAiOjE0NzUzODUzNTUsIm5iZiI6MTQ3NTM4MTc1NSwianRpIjoiYTE2NjY3ODA4NzA0ZTljM2NmNmZiZjMyMDM4NGIxY2YiLCJzdWIiOjF9.tz_DUHUaaD77zMkAk6cUd7-hUQYoQHlr3WGgDQZAupA\r\nContent-Length: 0\r\nHost: localhost:8000\r\nConnection: close\r\n\r\n',
[2]         _headers: [Object],
[2]         _headerNames: [Object],
[2]         _onPendingData: null,
[2]         agent: [Object],
[2]         socketPath: undefined,
[2]         method: 'GET',
[2]         path: '/api/v1/groups?draw=1&columns=%5Bobject%20Object%5D&columns=%5Bobject%20Object%5D&columns=%5Bobject%20Object%5D&columns=%5Bobject%20Object%5D&order=%5Bobject%20Object%5D&start=0&length=10&search=%5Bobject%20Object%5D&_=1475381757531',
[2]         _ended: true,
[2]         timeoutCb: [Function: emitTimeout],
[2]         parser: null,
[2]         res: [Circular] },
[2]      _idleTimeout: -1,
[2]      _idleNext: null,
[2]      _idlePrev: null,
[2]      _idleStart: 55893,
[2]      read: [Function],
[2]      _consuming: true,
[2]      write: [Function: writeAfterFIN] },
[2]   httpVersionMajor: 1,
[2]   httpVersionMinor: 1,
[2]   httpVersion: '1.1',
[2]   complete: true,
[2]   headers: 
[2]    { host: 'localhost:8000',
[2]      connection: 'close',
[2]      'x-powered-by': 'PHP/7.0.8-0ubuntu0.16.04.2',
[2]      'cache-control': 'no-cache, private',
[2]      authorization: 'Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwOi8vbG9jYWxob3N0OjgwMDAvYXBpL3YxL2dyb3VwcyIsImlhdCI6MTQ3NTM4MTc1NSwiZXhwIjoxNDc1Mzg1MzY4LCJuYmYiOjE0NzUzODE3NjgsImp0aSI6IjA0NDEyNTQxZTBkNDViNTIzMGJmZDRjMmM1YmRmMzRkIiwic3ViIjoxfQ.jsFayd41JRJk80l_BRERssf-yB5rZKcG6YPW6mgZw38',
[2]      'x-ratelimit-limit': '60',
[2]      'x-ratelimit-remaining': '59',
[2]      'content-type': 'text/html; charset=UTF-8',
[2]      date: 'Sun, 02 Oct 2016 04:16:08 GMT' },
[2]   rawHeaders: 
[2]    [ 'Host',
[2]      'localhost:8000',
[2]      'Connection',
[2]      'close',
[2]      'X-Powered-By',
[2]      'PHP/7.0.8-0ubuntu0.16.04.2',
[2]      'Cache-Control',
[2]      'no-cache, private',
[2]      'Authorization',
[2]      'Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwOi8vbG9jYWxob3N0OjgwMDAvYXBpL3YxL2dyb3VwcyIsImlhdCI6MTQ3NTM4MTc1NSwiZXhwIjoxNDc1Mzg1MzY4LCJuYmYiOjE0NzUzODE3NjgsImp0aSI6IjA0NDEyNTQxZTBkNDViNTIzMGJmZDRjMmM1YmRmMzRkIiwic3ViIjoxfQ.jsFayd41JRJk80l_BRERssf-yB5rZKcG6YPW6mgZw38',
[2]      'X-RateLimit-Limit',
[2]      '60',
[2]      'X-RateLimit-Remaining',
[2]      '59',
[2]      'Content-Type',
[2]      'text/html; charset=UTF-8',
[2]      'Date',
[2]      'Sun, 02 Oct 2016 04:16:08 GMT' ],
[2]   trailers: {},
[2]   rawTrailers: [],
[2]   upgrade: false,
[2]   url: '',
[2]   method: null,
[2]   statusCode: 500,
[2]   statusMessage: 'Internal Server Error',
[2]   client: 
[2]    Socket {
[2]      connecting: false,
[2]      _hadError: false,
[2]      _handle: null,
[2]      _parent: null,
[2]      _host: 'localhost',
[2]      _readableState: 
[2]       ReadableState {
[2]         objectMode: false,
[2]         highWaterMark: 16384,
[2]         buffer: [Object],
[2]         length: 0,
[2]         pipes: null,
[2]         pipesCount: 0,
[2]         flowing: true,
[2]         ended: true,
[2]         endEmitted: true,
[2]         reading: false,
[2]         sync: false,
[2]         needReadable: false,
[2]         emittedReadable: false,
[2]         readableListening: false,
[2]         resumeScheduled: false,
[2]         defaultEncoding: 'utf8',
[2]         ranOut: false,
[2]         awaitDrain: 0,
[2]         readingMore: false,
[2]         decoder: null,
[2]         encoding: null },
[2]      readable: false,
[2]      domain: null,
[2]      _events: 
[2]       { end: [Object],
[2]         finish: [Function: onSocketFinish],
[2]         _socketEnd: [Function: onSocketEnd],
[2]         free: [Function: onFree],
[2]         close: [Object],
[2]         agentRemove: [Function: onRemove],
[2]         drain: [Function: ondrain],
[2]         error: [Function: socketErrorListener],
[2]         data: [Function: socketOnData],
[2]         timeout: [Object] },
[2]      _eventsCount: 10,
[2]      _maxListeners: undefined,
[2]      _writableState: 
[2]       WritableState {
[2]         objectMode: false,
[2]         highWaterMark: 16384,
[2]         needDrain: false,
[2]         ending: true,
[2]         ended: true,
[2]         finished: true,
[2]         decodeStrings: false,
[2]         defaultEncoding: 'utf8',
[2]         length: 0,
[2]         writing: false,
[2]         corked: 0,
[2]         sync: false,
[2]         bufferProcessing: false,
[2]         onwrite: [Function],
[2]         writecb: null,
[2]         writelen: 0,
[2]         bufferedRequest: null,
[2]         lastBufferedRequest: null,
[2]         pendingcb: 0,
[2]         prefinished: true,
[2]         errorEmitted: false,
[2]         bufferedRequestCount: 0,
[2]         corkedRequestsFree: [Object] },
[2]      writable: false,
[2]      allowHalfOpen: false,
[2]      destroyed: true,
[2]      _bytesDispatched: 657,
[2]      _sockname: null,
[2]      _pendingData: null,
[2]      _pendingEncoding: '',
[2]      server: null,
[2]      _server: null,
[2]      parser: null,
[2]      _httpMessage: 
[2]       ClientRequest {
[2]         domain: null,
[2]         _events: [Object],
[2]         _eventsCount: 3,
[2]         _maxListeners: undefined,
[2]         output: [],
[2]         outputEncodings: [],
[2]         outputCallbacks: [],
[2]         outputSize: 0,
[2]         writable: true,
[2]         _last: true,
[2]         upgrading: false,
[2]         chunkedEncoding: false,
[2]         shouldKeepAlive: false,
[2]         useChunkedEncodingByDefault: false,
[2]         sendDate: false,
[2]         _removedHeader: [Object],
[2]         _contentLength: 0,
[2]         _hasBody: true,
[2]         _trailer: '',
[2]         finished: true,
[2]         _headerSent: true,
[2]         socket: [Circular],
[2]         connection: [Circular],
[2]         _header: 'GET /api/v1/groups?draw=1&columns=%5Bobject%20Object%5D&columns=%5Bobject%20Object%5D&columns=%5Bobject%20Object%5D&columns=%5Bobject%20Object%5D&order=%5Bobject%20Object%5D&start=0&length=10&search=%5Bobject%20Object%5D&_=1475381757531 HTTP/1.1\r\nServer-token: B@s~DSSwewaC63n0#xIf~9ROAoU$d$Z1WM|\r\nAuthorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwOi8vbG9jYWxob3N0OjgwMDAvYXBpL3YxL2xvZ2luIiwiaWF0IjoxNDc1MzgxNzU1LCJleHAiOjE0NzUzODUzNTUsIm5iZiI6MTQ3NTM4MTc1NSwianRpIjoiYTE2NjY3ODA4NzA0ZTljM2NmNmZiZjMyMDM4NGIxY2YiLCJzdWIiOjF9.tz_DUHUaaD77zMkAk6cUd7-hUQYoQHlr3WGgDQZAupA\r\nContent-Length: 0\r\nHost: localhost:8000\r\nConnection: close\r\n\r\n',
[2]         _headers: [Object],
[2]         _headerNames: [Object],
[2]         _onPendingData: null,
[2]         agent: [Object],
[2]         socketPath: undefined,
[2]         method: 'GET',
[2]         path: '/api/v1/groups?draw=1&columns=%5Bobject%20Object%5D&columns=%5Bobject%20Object%5D&columns=%5Bobject%20Object%5D&columns=%5Bobject%20Object%5D&order=%5Bobject%20Object%5D&start=0&length=10&search=%5Bobject%20Object%5D&_=1475381757531',
[2]         _ended: true,
[2]         timeoutCb: [Function: emitTimeout],
[2]         parser: null,
[2]         res: [Circular] },
[2]      _idleTimeout: -1,
[2]      _idleNext: null,
[2]      _idlePrev: null,
[2]      _idleStart: 55893,
[2]      read: [Function],
[2]      _consuming: true,
[2]      write: [Function: writeAfterFIN] },
[2]   _consuming: true,
[2]   _dumped: false,
[2]   req: 
[2]    ClientRequest {
[2]      domain: null,
[2]      _events: 
[2]       { timeout: [Object],
[2]         error: [Function],
[2]         prefinish: [Function: requestOnPrefinish] },
[2]      _eventsCount: 3,
[2]      _maxListeners: undefined,
[2]      output: [],
[2]      outputEncodings: [],
[2]      outputCallbacks: [],
[2]      outputSize: 0,
[2]      writable: true,
[2]      _last: true,
[2]      upgrading: false,
[2]      chunkedEncoding: false,
[2]      shouldKeepAlive: false,
[2]      useChunkedEncodingByDefault: false,
[2]      sendDate: false,
[2]      _removedHeader: { 'content-length': false },
[2]      _contentLength: 0,
[2]      _hasBody: true,
[2]      _trailer: '',
[2]      finished: true,
[2]      _headerSent: true,
[2]      socket: 
[2]       Socket {
[2]         connecting: false,
[2]         _hadError: false,
[2]         _handle: null,
[2]         _parent: null,
[2]         _host: 'localhost',
[2]         _readableState: [Object],
[2]         readable: false,
[2]         domain: null,
[2]         _events: [Object],
[2]         _eventsCount: 10,
[2]         _maxListeners: undefined,
[2]         _writableState: [Object],
[2]         writable: false,
[2]         allowHalfOpen: false,
[2]         destroyed: true,
[2]         _bytesDispatched: 657,
[2]         _sockname: null,
[2]         _pendingData: null,
[2]         _pendingEncoding: '',
[2]         server: null,
[2]         _server: null,
[2]         parser: null,
[2]         _httpMessage: [Circular],
[2]         _idleTimeout: -1,
[2]         _idleNext: null,
[2]         _idlePrev: null,
[2]         _idleStart: 55893,
[2]         read: [Function],
[2]         _consuming: true,
[2]         write: [Function: writeAfterFIN] },
[2]      connection: 
[2]       Socket {
[2]         connecting: false,
[2]         _hadError: false,
[2]         _handle: null,
[2]         _parent: null,
[2]         _host: 'localhost',
[2]         _readableState: [Object],
[2]         readable: false,
[2]         domain: null,
[2]         _events: [Object],
[2]         _eventsCount: 10,
[2]         _maxListeners: undefined,
[2]         _writableState: [Object],
[2]         writable: false,
[2]         allowHalfOpen: false,
[2]         destroyed: true,
[2]         _bytesDispatched: 657,
[2]         _sockname: null,
[2]         _pendingData: null,
[2]         _pendingEncoding: '',
[2]         server: null,
[2]         _server: null,
[2]         parser: null,
[2]         _httpMessage: [Circular],
[2]         _idleTimeout: -1,
[2]         _idleNext: null,
[2]         _idlePrev: null,
[2]         _idleStart: 55893,
[2]         read: [Function],
[2]         _consuming: true,
[2]         write: [Function: writeAfterFIN] },
[2]      _header: 'GET /api/v1/groups?draw=1&columns=%5Bobject%20Object%5D&columns=%5Bobject%20Object%5D&columns=%5Bobject%20Object%5D&columns=%5Bobject%20Object%5D&order=%5Bobject%20Object%5D&start=0&length=10&search=%5Bobject%20Object%5D&_=1475381757531 HTTP/1.1\r\nServer-token: B@s~DSSwewaC63n0#xIf~9ROAoU$d$Z1WM|\r\nAuthorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwOi8vbG9jYWxob3N0OjgwMDAvYXBpL3YxL2xvZ2luIiwiaWF0IjoxNDc1MzgxNzU1LCJleHAiOjE0NzUzODUzNTUsIm5iZiI6MTQ3NTM4MTc1NSwianRpIjoiYTE2NjY3ODA4NzA0ZTljM2NmNmZiZjMyMDM4NGIxY2YiLCJzdWIiOjF9.tz_DUHUaaD77zMkAk6cUd7-hUQYoQHlr3WGgDQZAupA\r\nContent-Length: 0\r\nHost: localhost:8000\r\nConnection: close\r\n\r\n',
[2]      _headers: 
[2]       { 'server-token': 'B@s~DSSwewaC63n0#xIf~9ROAoU$d$Z1WM|',
[2]         authorization: 'Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwOi8vbG9jYWxob3N0OjgwMDAvYXBpL3YxL2xvZ2luIiwiaWF0IjoxNDc1MzgxNzU1LCJleHAiOjE0NzUzODUzNTUsIm5iZiI6MTQ3NTM4MTc1NSwianRpIjoiYTE2NjY3ODA4NzA0ZTljM2NmNmZiZjMyMDM4NGIxY2YiLCJzdWIiOjF9.tz_DUHUaaD77zMkAk6cUd7-hUQYoQHlr3WGgDQZAupA',
[2]         'content-length': 0,
[2]         host: 'localhost:8000' },
[2]      _headerNames: 
[2]       { 'server-token': 'Server-token',
[2]         authorization: 'Authorization',
[2]         'content-length': 'Content-Length',
[2]         host: 'Host' },
[2]      _onPendingData: null,
[2]      agent: 
[2]       Agent {
[2]         domain: null,
[2]         _events: [Object],
[2]         _eventsCount: 1,
[2]         _maxListeners: undefined,
[2]         defaultPort: 80,
[2]         protocol: 'http:',
[2]         options: [Object],
[2]         requests: {},
[2]         sockets: [Object],
[2]         freeSockets: {},
[2]         keepAliveMsecs: 1000,
[2]         keepAlive: false,
[2]         maxSockets: Infinity,
[2]         maxFreeSockets: 256 },
[2]      socketPath: undefined,
[2]      method: 'GET',
[2]      path: '/api/v1/groups?draw=1&columns=%5Bobject%20Object%5D&columns=%5Bobject%20Object%5D&columns=%5Bobject%20Object%5D&columns=%5Bobject%20Object%5D&order=%5Bobject%20Object%5D&start=0&length=10&search=%5Bobject%20Object%5D&_=1475381757531',
[2]      _ended: true,
[2]      timeoutCb: [Function: emitTimeout],
[2]      parser: null,
[2]      res: [Circular] },
[2]   read: [Function] }
```

console.log(dataClient.parameters)

```
[2] [dev:server] { draw: '1',
[2]   columns: 
[2]    [ { data: '0',
[2]        name: '',
[2]        searchable: 'true',
[2]        orderable: 'true',
[2]        search: [Object] },
[2]      { data: '1',
[2]        name: '',
[2]        searchable: 'true',
[2]        orderable: 'true',
[2]        search: [Object] },
[2]      { data: '2',
[2]        name: '',
[2]        searchable: 'true',
[2]        orderable: 'true',
[2]        search: [Object] },
[2]      { data: '3',
[2]        name: '',
[2]        searchable: 'true',
[2]        orderable: 'true',
[2]        search: [Object] } ],
[2]   order: [ { column: '0', dir: 'asc' } ],
[2]   start: '0',
[2]   length: '10',
[2]   search: { value: '', regex: 'false' },
[2]   _: '1475381757531' }
```

Question: why path is 

```
/api/v1/groups?draw=1&columns=%5Bobject%20Object%5D&columns=%5Bobject%20Object%5D&columns=%5Bobject%20Object%5D&columns=%5Bobject%20Object%5D&order=%5Bobject%20Object%5D&start=0&length=10&search=%5Bobject%20Object%5D&_=1475381757531
```

although it should be equal

```
/api/v1/groups?draw=1&columns%5B0%5D%5Bdata%5D=0&columns%5B0%5D%5Bname%5D=&columns%5B0%5D%5Bsearchable%5D=true&columns%5B0%5D%5Borderable%5D=true&columns%5B0%5D%5Bsearch%5D%5Bvalue%5D=&columns%5B0%5D%5Bsearch%5D%5Bregex%5D=false&columns%5B1%5D%5Bdata%5D=1&columns%5B1%5D%5Bname%5D=&columns%5B1%5D%5Bsearchable%5D=true&columns%5B1%5D%5Borderable%5D=true&columns%5B1%5D%5Bsearch%5D%5Bvalue%5D=&columns%5B1%5D%5Bsearch%5D%5Bregex%5D=false&columns%5B2%5D%5Bdata%5D=2&columns%5B2%5D%5Bname%5D=&columns%5B2%5D%5Bsearchable%5D=true&columns%5B2%5D%5Borderable%5D=true&columns%5B2%5D%5Bsearch%5D%5Bvalue%5D=&columns%5B2%5D%5Bsearch%5D%5Bregex%5D=false&columns%5B3%5D%5Bdata%5D=3&columns%5B3%5D%5Bname%5D=&columns%5B3%5D%5Bsearchable%5D=true&columns%5B3%5D%5Borderable%5D=true&columns%5B3%5D%5Bsearch%5D%5Bvalue%5D=&columns%5B3%5D%5Bsearch%5D%5Bregex%5D=false&order%5B0%5D%5Bcolumn%5D=0&order%5B0%5D%5Bdir%5D=asc&start=0&length=10&search%5Bvalue%5D=&search%5Bregex%5D=false&_=1475380958097
```

**Comment by aacerox** [comment]
Hi

in your case your passing an object as parameter and want that this object would be serialized as is, while "parameters" arg is only intended to be used as a container for replacement variables for parameter string in a URL.

Currently I'm working in a improvement that will allow what you need: serialize and encode a js object as a parameter URL string.

It will be available asap.

thanks.

**Comment by aacerox** [comment]
Hi

in your case you're passing an object as parameter and want that this object would be serialized as a parameter string in the URL, while "parameters" argument is only intended to be used as a container for replacement variables for parameter string in a URL.

Currently I'm working in a improvement that will allow what you need: serialize and encode a js object as a parameter URL string.

It will be available asap.

thanks.

**Comment by aacerox** [comment]
forget what I said... it's intended to work as you need....
Big Bad Bug
already fixing it...

**Comment by aacerox** [comment]
Hi
this should be fixed in version 3.0, please give it a try and tell me if it works.

thnx
